src/index.ts

```diff
@@ -38,7 +39,11 @@ export const monitorCharacteristic = (
       const value = Buffer.from(rawData.buffer, rawData.byteOffset, rawData.byteLength);
       subscriber.next(value);
     })
-  });
+  }).pipe(
+    finalize(() => {
+      BleClient.stopNotifications(deviceId, service, characteristic);
```

Comment:
The Rx idiomatic way to do it is to return dispose function inside `new Observable(`

Reply:
You probably mean `subscriber.complete()`. But in this case I want to terminate the subscription on device disconnect. Which is out of context in this scope. `finalize` does exactly that, it is a side effect of complete observable.

Reply:
I don't understand. I mean:
```ts
  return new Observable<Buffer>(subscriber => {
    BleClient.startNotifications(deviceId, service, characteristic, rawData => {
      const value = Buffer.from(rawData.buffer, rawData.byteOffset, rawData.byteLength);
      subscriber.next(value);
    });
    return () => {
      BleClient.stopNotifications(deviceId, service, characteristic);
    }; 
  });
```

Docs: https://rxjs.dev/guide/observable#disposing-observable-executions

Reply:
I see what you mean now. Should work the similar way.